cmds/mkdir/mkdir.go

```diff
@@ -16,32 +16,93 @@ package main
 import (
 	"flag"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
+	"syscall"
+)
+
+type modeFlag struct {
```

Comment:
I don't understand the need for this struct.

Reply:
So in mkdir, when you do not set the -m flag, they will use your default creation mode & ~umask to be the creation mode of your directory. However, when you set the -m flag, the umask is effectively set to 0 before the creation mode is taken into consideration. Since flag.String in Go has no way of indicating whether a flag is set or not, because they only return a value, we need to create our own flag for that purpose.

Reply:
I'll stop by.

Reply:
you can actually test this pretty easily. 
m = flag.String("m", "", "creation mask")

then later:
if *m == "" {
  *m = "0777"
} else {
oldUmask = umask(0)
defer umask(oldUmask)
}

Then further down
mode := strconv.whatever(*m, 8, whatever)

an unset string has value ""

Reply:
Fixed